CONTRIBUTING.md

```diff
@@ -0,0 +1,145 @@
+
+* [Technical Mailing List](http://lists.projectcalico.org/mailman/listinfo/calico-tech_lists.projectcalico.org)
+* [Slack Calico Users Channel](https://calicousers.slack.com) ([Sign up](https://calicousers-slackin.herokuapp.com))
+* IRC - [#calico](https://kiwiirc.com/client/irc.freenode.net/#calico)
+
+## Reporting issues
+
+Before raising an issue with the *Calico Libnetwork plugin*), please check for 
+duplicate issues, and read our [Troubleshooting](https://github.com/projectcalico/calico-containers/blob/master/docs/Troubleshooting.md)
+and our [Frequently Asked Questions](https://github.com/projectcalico/calico-containers/blob/master/docs/FAQ.md)
+documents.  You may also find helpful information in our [guides on Calico as a Docker network plugin]
+(https://github.com/projectcalico/calico-containers/tree/master/docs/calico-with-docker/docker-network-plugin).
+
+If you have a question, please hop on to our IRC or Slack channels (see above).
+
+If you do need to raise an issue, please include any of the following that may
+be relevant to your problem (including too much information is never
+a bad thing):
+
+-  A detailed description of the problem.
+-  The version of Docker in use from the `docker version` command.
+-  The steps to reproduce the problem (e.g. the full list of `calicoctl`
+   commands that were run)
+-  The expected result and actual result.
+-  Versions of appropriate binaries and libraries.  For example, the output from
+   `calicoctl version`, your version of Mesos, rkt, Kubernetes etc.
+-  A link to any diagnostics (e.g. if using `calicoctl`, you can gathered
+   diagnostics using `calicoctl diags` - this provides instructions for 
+   uploading the diags bundle to transfer.sh - or alternatively if the 
+   diagnostics contains sensitive information we can set up an alternative
+   method for transfer).
+-  If using `calicoctl` the output from `calicoctl status` run on each node
+   might also be useful.
+-  Details of your OS.
+-  Environment details such as GCE, bare metal, VirtualBox.
+   
+
+## Contributing code and documentation
+
+For contributing code and documentation we follow the GitHub pull request
+model. 
+
+-  Fork the repository on GitHub
+-  Make changes to your local repository in a separate branch
+-  Test the change
+-  Create a pull request which will then go through a review process by one or
+   more of the core team.
+
+### Testing your changes
+
+If you create a pull request, our automated UT and STs will be run over your
+change.  We will not accept changes that do not consistently pass our automated
+test suites. It is vital that our master branch be passing tests at all times.
+If you tests are failing the automated tests and you don't believe they should
+be, you may need to rebase your branch off the latest master.
+
+The unit tests can be run with:
+```
+make test
+```
+
+Where possible, please add any additional tests to ensure we maintain healthy
+code and feature coverage.
```

Comment:
I would also point at the main README for details on how to run the plugin for live testing.